category/category.go

```diff
@@ -0,0 +1,287 @@
+func (m *GormRepository) CreateRelationship(ctx context.Context, relationship *WorkItemTypeCategoryRelationship) error {
+	db := m.db.Create(relationship)
+	if db.Error != nil {
+		if gormsupport.IsUniqueViolation(db.Error, "work_item_type_categories_idx") {
+			return errors.NewBadParameterError("category+workitemtype", relationship.CategoryID).Expected("unique")
+		}
+		log.Error(ctx, map[string]interface{}{
+			"category_id":     relationship.CategoryID,
+			"workitemtype_id": relationship.WorkitemtypeID,
+			"err":             db.Error.Error(),
+		}, "unable to create workitemtype category relationship")
+		return errors.NewInternalError(db.Error.Error())
+	}
+	return nil
+}
+
+// Create creates category. This function is used to populate categories table during migration -> PopulateCategories()
+func (m *GormRepository) Create(ctx context.Context, category *Category) (*Category, error) {
+	if strings.TrimSpace(category.Name) == "" {
+		return nil, errors.NewBadParameterError("Name", category.Name)
+
+	}
+	db := m.db.Create(category)
+	if db.Error != nil {
+		if gormsupport.IsUniqueViolation(db.Error, "categories_name_idx") {
+			return nil, errors.NewBadParameterError("Name", category.Name).Expected("unique")
+		}
+		log.Error(ctx, map[string]interface{}{
+			"category_id": category.ID,
+			"err":         db.Error.Error(),
+		}, "unable to create category")
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	log.Info(ctx, map[string]interface{}{
+		"category_id": category.ID,
+	}, "category created successfully")
+	return category, nil
+}
+
+// LoadAllRelationshipsOfCategory loads all the relationships of a category. This is required for workitemtype filtering.
+func (m *GormRepository) LoadAllRelationshipsOfCategory(ctx context.Context, categoryID uuid.UUID) ([]*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	getCategory := Category{}
+	db := m.db.Model(&getCategory).Where("id=?", categoryID).Find(&getCategory)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+		}, "category not found")
+		return nil, errors.NewNotFoundError("category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+			"err":         err,
+		}, "unable to list relationships")
+		return nil, errors.NewInternalError(err.Error())
+	}
+	relationship := []*WorkItemTypeCategoryRelationship{}
+	db = m.db.Model(&relationship).Where("category_id=?", categoryID).Find(&relationship)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+		}, "workitemtypes of category not found")
+		return nil, errors.NewNotFoundError("work item type category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+			"err":         err,
+		}, "workitemtypes of category not found")
```

Comment:
this error message is hiding the current error. Use a more appropriated message, but don't use `not found`